src/Microsoft.VisualStudio.SlowCheetah/Transformer/XmlTransformer.cs

```diff
@@ -21,59 +22,121 @@ public XmlTransformer()
         {
         }
 
-        /// <summary>
-        /// Initializes a new instance of the <see cref="XmlTransformer"/> class with an external logger.
-        /// </summary>
-        /// <param name="logger">External logger. Passed into the transformation</param>
-        /// <param name="useSections">Wheter or not to use sections while logging</param>
-        public XmlTransformer(ITransformationLogger logger, bool useSections)
+        /// <inheritdoc/>
+        public void CreateTransformFile(string sourcePath, string transformPath, bool overwrite)
+        {
+            if (string.IsNullOrWhiteSpace(sourcePath))
+            {
+                throw new ArgumentNullException(nameof(sourcePath));
+            }
+
+            if (string.IsNullOrWhiteSpace(transformPath))
+            {
+                throw new ArgumentNullException(nameof(transformPath));
+            }
+
+            if (!File.Exists(sourcePath))
+            {
+                throw new FileNotFoundException(Resources.Resources.ErrorMessage_SourceFileNotFound, sourcePath);
+            }
+
+            // If the file should be overwritten or if it doesn't exist, we create it
+            if (overwrite || !File.Exists(transformPath))
+            {
+                // First, copy the original file to preserve encoding and header
+                File.Copy(sourcePath, transformPath, true);
```

Comment:
Use the `overwrite` variable. `File.Copy(sourcePath, transformPath, overwrite);`